src/coreclr/src/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/FieldAccessors/WritableStaticFieldAccessor.cs

```diff
@@ -16,14 +16,26 @@ protected WritableStaticFieldAccessor(IntPtr cctorContext, RuntimeTypeHandle fie
 
         protected abstract override object GetFieldBypassCctor();
 
+        protected abstract override bool IsFieldInitOnly { get; }
+
         protected sealed override void SetFieldBypassCctor(object value, BinderBundle binderBundle)
         {
+            if (IsFieldInitOnly)
```

Comment:
Could you move the CheckArgument call before this line?

(Same for the Direct version below)